adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -463,17 +474,103 @@ private void handleConnectionRequestResult(final MqttEndpoint endpoint,
 
     private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device authenticatedDevice) {
 
+        commandSubscription(endpoint, authenticatedDevice);
+
         endpoint.closeHandler(v -> {
             close(endpoint, authenticatedDevice);
-            LOG.debug("connection to device [tenant-id: {}, device-id: {}] closed",
-                    authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId());
-            metrics.decrementMqttConnections(authenticatedDevice.getTenantId());
         });
 
         endpoint.publishHandler(message -> handlePublishedMessage(new MqttContext(message, endpoint, authenticatedDevice)));
         metrics.incrementMqttConnections(authenticatedDevice.getTenantId());
     }
 
+    private void commandSubscription(final MqttEndpoint endpoint, final Device authenticatedDevice) {
```

Comment:
`commandSubscription` is just a noun. Please prepend a verb so that it becomes (more) obvious what this method does.